tests/check/check_test.go

```diff
@@ -53,9 +54,12 @@ func testCheck(t *testing.T, engine string) {
 	)
 	require.NoError(t, err)
 	defer conn.Close()
+	t.Run("testCheck", func(t *testing.T) {
+
+		RunSchema1_1CheckTests(t, pb.NewOpenFGAServiceClient(conn))
```

Comment:
These should be parallelized as well. Right now they are running sequentially.

This is why I suggested we refactor a bit and move these up a level (see https://github.com/openfga/openfga/pull/584#discussion_r1122058504).

I think what we want, for consistency and the most parallelism, is something like the following:

```
func TestCheckMemory(t *testing.T) {
    RunAllCheckTests(t, "memory")
}

func RunAllCheckTests(t *testing.T, engine string) {
    t.Run("Check", func(t *testing.T) { t.Parallel(); checkTest(t, engine) }
    t.Run("BadAuthModelID", func(t *testing.T) { t.Parallel(); badAuthModelIDTest(t, engine) }
}

func checkTest(t *testing.T, engine string) {
    ... // boilerplate setup

    t.Run("SchemaVersion1_0", func(t *testing.T) {
    
    })

   t.Run("SchemaVersion1_1", func(t *testing.T) {
   
   })
}
```

This will make our tests more parallel and will also make them more consistent with other tests in this codebase. For example:
https://github.com/openfga/openfga/blob/main/pkg/server/test/server.go

The pattern we've been trying to follow lately (though we need to refactor some) is to name test signatures of the form `func somethingTest(t *testing.T, ... args)` and reserve function names like `TestSomething` for actual go test signatures (e.g. `TestSomething(t *testing.T)`).